crates/agglayer-gcp-kms/src/signer.rs

```diff
@@ -31,39 +30,96 @@ impl KmsSigner {
         &self,
         message: S,
     ) -> Result<Signature, Error> {
-        Ok(self.signer.sign_message(message).await?)
+        self.signer
+            .sign_message(message.as_ref())
+            .await
+            .map_err(|e| Error::KmsError(e.to_string()))
     }
 
     /// Signs a transaction using the internal signer, this method can fail if
     /// the signer fails to create the digest.
     pub async fn sign_transaction(&self, tx: &TypedTransaction) -> Result<Signature, Error> {
-        Ok(self.signer.sign_transaction(tx).await?)
-    }
-
-    /// Signs typed data using internal signer.
-    ///
-    /// This method can fail while trying to encode the payload using EIP-712 or
-    /// during the digest creation.
-    pub async fn sign_typed_data<T: Eip712 + Send + Sync>(
-        &self,
-        payload: &T,
-    ) -> Result<Signature, Error> {
-        Ok(self.signer.sign_typed_data(payload).await?)
+        // Convert the TypedTransaction to a mutable dyn SignableTransaction
+        let mut tx_clone = tx.clone();
+        self.signer
+            .sign_transaction(&mut tx_clone)
+            .await
+            .map_err(|e| Error::KmsError(e.to_string()))
     }
 
     /// Returns the address associated with the signer.
     pub fn address(&self) -> Address {
-        self.signer.address()
+        alloy::signers::Signer::address(&self.signer)
     }
 
     /// Returns the chain ID associated with the signer.
-    pub fn chain_id(&self) -> u64 {
+    pub fn chain_id(&self) -> Option<ChainId> {
         self.signer.chain_id()
     }
 
     /// Sets a new chain ID for the signer.
     pub fn with_chain_id<T: Into<u64>>(mut self, chain_id: T) -> Self {
-        self.signer = self.signer.with_chain_id(chain_id);
+        self.signer = self.signer.with_chain_id(Some(chain_id.into()));
         self
     }
+
+    /// Sets the chain ID on this signer (mutable version).
+    pub fn set_chain_id(&mut self, chain_id: Option<ChainId>) {
+        if let Some(chain_id) = chain_id {
+            self.signer = self.signer.clone().with_chain_id(Some(chain_id));
+        }
+    }
```

Comment:
This will leave the chain ID unchanged rather than setting it to `None` when `None` is passed in. Is that intentional?

Reply:
Yes, makes more sense for user to decide if he wants to set `None` here.